### DISCUSSION TITLE
SendKeyEvent with Alt down

### DISCUSSION THREAD

**Discussion by kevinvenclovas** [cefsharp/CefSharp#4206] [General]
Hi, 

i want to send System.Windows.Forms.Keys.NumPad1 with alt pressed to the browser.

Key: ALT + NUM1

i use this code but the browser recieved always only NUM1 without ALT pressed. Someone can help me here? 

```
KeyEvent keyEvent = new KeyEvent();
keyEvent.WindowsKeyCode = (int)System.Windows.Forms.Keys.NumPad1;
keyEvent.FocusOnEditableField = true;
keyEvent.IsSystemKey = false;
keyEvent.Modifiers = CefEventFlags.AltDown;
keyEvent.Type = KeyEventType.KeyDown;
browser.GetBrowser().GetHost().SendKeyEvent(keyEvent);
```

**Comment by amaitland** [comment]
If I remember correctly alt is a system key. Not sure if your modifiers are correct either.  There's reference code available at https://github.com/chromiumembedded/cef/blob/cb83a7c3fddacb984df60d303701a3d25ad1f826/tests/cefclient/browser/osr_window_win.cc#L783

https://docs.microsoft.com/en-us/windows/win32/inputdev/wm-syskeydown

Usually I'd download https://cef-builds.spotifycdn.com/cef_binary_104.4.22%2Bg181df42%2Bchromium-104.0.5112.81_windows32.tar.bz2 build the cefclient reference example and debug to see exactly what's required. 

---

Depending what you are actually trying to achieve using https://github.com/cefsharp/CefSharp.Dom is probably an easier option, it provides a higher level keyboard API. 

- https://github.com/cefsharp/CefSharp.Dom/blob/main/lib/PuppeteerSharp.Tests/KeyboardTests/KeyboardTests.cs#L144
- https://github.com/cefsharp/CefSharp.Dom/blob/main/lib/PuppeteerSharp.Tests/KeyboardTests/KeyboardTests.cs#L294
- https://github.com/cefsharp/CefSharp.Dom/blob/main/lib/PuppeteerSharp/Input/KeyDefinitions.cs#L255
